### PR TITLE
misc: add issue templates

### PR DESCRIPTION
**Pull request by rfratto**
Add issue templates for:

* Bug reports, to allow users to report specific problems.
* Feature requests, to allow users to request that new functionality be added.
* Proposals, to allow users to propose a specific implementation for a problem they would like to solve.

Feature requests are just off-the-cuff suggestions, while proposals are expected to go more in depth about the background and the proposed solution(s).

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+        Please do not use GitHub issues for support questions. If you need help or support, ask a question in [GitHub Discussions](https://github.com/grafana/agent/discussions) or join the `#agent` channel on the [Grafana Slack](https://slack.grafana.com/).
+  - type: textarea
+    attributes:
+      label: What's wrong?
+      description: Provide a brief description of the issue.
+  - type: textarea
+    attributes:
+      label: Steps to reproduce
+      descriptions: Please describe steps for us to be able to reproduce the issue.
```
Comment (mattdurham) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:17]: Should this be `description`?

Comment (rfratto) [reply .github/ISSUE_TEMPLATE/bug_report.yaml:17]: It sure should be, thanks

Comment (rfratto) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:17]:
```suggestion
      description: Please describe steps for us to be able to reproduce the issue.
```

File: .github/ISSUE_TEMPLATE/bug_report.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      placeholder: e.g. Grafana Agent v0.33.0
+  - type: textarea
+    attributes:
+      label: Configuration
+      description: Provide the relevant configuration file here. Remove sensitive information before sharing.
+      render: text
+  - type: textarea
+    attributes:
+      label: Logs
+      description: Provide logs relevant to the issue.
```
Comment (mattdurham) [inline .github/ISSUE_TEMPLATE/bug_report.yaml:38]: Add disclaimer to remove sensitive information?

File: .github/ISSUE_TEMPLATE/proposal.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+name: Proposal
+description: Propose a way to solve a problem.
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thank you for opening a bug report for Grafana Agent.
```
Comment (mattdurham) [inline .github/ISSUE_TEMPLATE/proposal.yaml:7]: `bug report` -> `proposal`?

File: .github/ISSUE_TEMPLATE/feature_request.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+name: Feature request
+description: Suggest an idea to be added to Grafana Agent.
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thank you for opening a bug report for Grafana Agent.
```
Comment (mattdurham) [inline .github/ISSUE_TEMPLATE/feature_request.yaml:7]: `bug report` -> `feature request`?

**Review by jcreixell — APPROVED** [review]

**Pull request merged**